src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -307,10 +318,13 @@ RTPSParticipantImpl::RTPSParticipantImpl(
 
     mp_builtinProtocols = new BuiltinProtocols();
 
-    //Start reception
-    for (auto& receiver : m_receiverResourcelist)
+    if (enabled)
     {
-        receiver.Receiver->RegisterReceiver(receiver.mp_receiver);
+        //Start reception
+        for (auto& receiver : m_receiverResourcelist)
+        {
+            receiver.Receiver->RegisterReceiver(receiver.mp_receiver);
+        }
     }
```

Comment:
This whole block can be removed, as it is done inside `enable()`. This also means that we can remove the `enabled` argument